nixos/modules/services/backup/syncoid.nix

```diff
@@ -0,0 +1,129 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.syncoid;
+in {
+
+    # Interface
+
+    options.services.syncoid = {
+      enable = mkEnableOption "Syncoid ZFS synchronization service";
+
+      interval = mkOption {
+        type = types.str;
+        default = "hourly";
+        example = "*-*-* *:15:00";
+        description = ''
+          Run syncoid at this interval. The default is to run hourly.
+
+          The format is described in
+          <citerefentry><refentrytitle>systemd.time</refentrytitle>
+          <manvolnum>7</manvolnum></citerefentry>.
+        '';
+      };
+
+      user = mkOption {
+        type = types.str;
+        default = "root";
+        example = "backup";
+        description = ''
+          The user for the service. ZFS privilege delegation must be configured
+          to use a user other than root.
+        '';
+      };
+
+      sshKey = mkOption {
+        type = types.nullOr types.path;
+        default = null;
+        description = ''
+          SSH private key file to use to login to remote system. Can be
+          overridden in individual commands.
+        '';
+      };
+
+      defaultArguments = mkOption {
```

Comment:
How about removing this option and `useDefaultArguments` and instead putting the default arguments as `default = [ ... ]` into `extraArguments`?

Reply:
The name was not a good description of what that the option does. I changed it to `commonArgs` (it seems like `Args` is more common than `Arguments` in nixpkgs). I created that option because I wanted a convenient shortcut to apply arguments to all (or most, since you can disable it for a specific command with `useCommonArgs = false`) of the commands, in addition to those configured in `extraArgs`. For example, I use it to add `--no-privilege-elevation` and `--no-sync-snap` in my configuration.